File: CSharpPractice/LeetcodePractice/Leetcode386.cs
public class Solution386 {
    public IList<int> LexicalOrder(int n) {
        List<int> ans = new List<int>();
        for (var i = 1; i < n+1; i++)
        {
            ans.Add(i);
        }
        ans = ans.OrderBy(x=>x.ToString()).ToList();

        // ans.Sort((a,b)=>a.ToString().CompareTo(b.ToString())); It also works

        return ans;
    }
}

File: CppPractice/CppLeetcode/Leetcode386.cpp
#include <vector>
class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> ans;
        for(int i = 1; i < n+1; i++)
        {
            ans.push_back(i);
        }
        // Sort the vector in ascending order
        sort(ans.begin(), ans.end(),[](int a,int b){
            return to_string(a)<to_string(b);
        });

        return ans;
    }
};